lesson5/main.js

```diff
@@ -0,0 +1,30 @@
+"use strict";
+
+const ul = document.getElementById("js-lists");
+const fragment = document.createDocumentFragment();
+const listItems = [
+  { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" },
+];
+
+const p = new Promise((resolve) => {
+  resolve(listItems);
+}).then((val) => {
```

Comment:
This 10 line to end line,

Here, I think can be create some functions.

Specifically,

①a function to create list (but, returning fragment)
②a function to insert ul created list by ①.
③a function to return resolved data.

I think it's a little difficult, 
so please ask again if you get stuck.

Reply:
I see. I thought you meant if it change to a function, we can use this in various situations. So I tried!
And also I didn't write "reject" in the function of Promise object, so I added.

[3d37d3f](https://github.com/yuka830/js-lessons/pull/7/commits/3d37d3f8562ff16b69ba9810c8063f68ba40e736)

Reply:
The following three processing can be divided into functions or variables.

```
new Promise((resolve, reject) => {
  resolve(listItems);
})
```
↑ Please refer to this Link.
（The ideal is looking like the sample code. reject is OK. Its fine as is.）
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Promise/then

```
data.forEach((key) => {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");

    a.href = `/${key.to}`;
    a.textContent = key.text;
    img.src = key.img;
    img.alt = key.alt;

    fragment
      .appendChild(li)
      .appendChild(a)
      .insertAdjacentElement("afterbegin", img);
});
```
↑ ①：This processing can be function as returning `fragment` .

```
const ul = document.getElementById("js-lists");
ul.appendChild(fragment);
```
↑ ②：set created by ① at args. 
And This also can also be created as a function.

Reply:
It's a little close!
The answer will be as per Discord.

I've written some code that is close to the intent of the question.
Please take a look at it if it's a giveaway.

https://codesandbox.io/s/lesson5-forked-kxcvz?file=/main.js

↑ But, I've left some room for modification yet.
Currently, variables are declared globally, but is it really necessary to do so?
Please think about it once.

I'm sorry. I may have asked for something difficult at the timing of task 5.

Reply:
MUST
@rik9228
Thank you review.

I think that Fragments look like global variable. Hence, It's don't need to return it. `insert List` don't need execute.
I think the code will end up like this. 
https://codesandbox.io/s/lesson5-forked-3uo42?file=/main.js

In addition,
`datas` -> `data`
data is plurals.

Reply:
> I think that Fragments look like global variable.
I'm sorry, but I don't understand much here.
Sorry, This mean is conventional thing in Javascript?

This is all the code I had in mind.

https://codesandbox.io/s/lesson5-forked-koxvc?file=/main.js

By doing this
・You don't have to declare variables globally.
・It looks cleaner because the processes are clearly separated.

I'm thinking about it but, How about it?

Reply:
this is ok, If you show it like this from the beginning, my point is unnecessary.

arguments change from `data` to `fragment`

Reply:
@rik9228 , @kenmori 
Thank you for the comments.
I modified my code with @rik9228 's advice. I implemented function of insertList and fix newList.
[6ae3289](https://github.com/yuka830/js-lessons/pull/7/commits/6ae32894c4f86c72541fcc7ba292d0e4cf97f200)
[CodeSandbox](https://codesandbox.io/s/lesson5fixing-by-myself-nrx67?file=/main.js)

I didn't have any ideas about organizing each function and using in various situations in the future.
And I compared mine to yours, then I realized something wrong in my code.
・the constants of fragment and list were not organized
・the way writing arrow function(on the last line)
So I fixed them with @rik9228 's code.
[4befc43](https://github.com/yuka830/js-lessons/pull/7/commits/4befc4397f194e17dd80b0933d31faeaa9c41a23)

Reply:
Good.
sorry, lastly I was wondering about the naming, please confirm!